### PR TITLE
Add envconfig to project

### PR DESCRIPTION
**Pull request by rafaelhenrique**
Trying solve issue #6 :smile:

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -15,17 +15,26 @@ import (
...
-		log.Fatal("Missing: PORT environment variable")
+	var env Environment
+	err := envconfig.Process("hercules", &env)
+	if err != nil {
+		log.Fatal("Configuration error:", err.Error())
+	}
+
+	// Grab $HERCULES_PORT from env
+	if env.Port == "" {
+		log.Fatal("Missing: HERCULES_PORT environment variable")
```
Comment (hypnoglow) [inline server.go]: You can use `required:"true"` tag on a struct field instead.

**Comment by kshitij10496** [comment]
This looks good @rafaelhenrique 😄 🎉 

Just a few suggestions to make this ready for merge:
1. As @hypnoglow mentioned above, we can provide a `required` flag to enforce that both the variables are set. 👏 
2. Use a default value for `PORT`(say 8080) which can be simply set using the `default` tag.
3. I would appreciate it very much if you can add a description using the `desc` tag for each of the variables.
4. Define the `Config` struct in `common` package and import it in `server.go`.
5. Process the input as:
```go
if err := envconfig.Process("hercules", &config); err != nil {
		fmt.Fprintln(os.Stderr, err)
		envconfig.Usage("hercules", &config)
		os.Exit(1)
	}
```

**Comment by rafaelhenrique** [comment]
Applied changes @kshitij10496 @hypnoglow ... thanks for tips :wink:

**Comment by kshitij10496** [comment]
Great work @rafaelhenrique! 🎉 

I will wait for @hypnoglow's review before merging it in. 😄

File: server.go [hunk 2]
```diff
@@ -16,16 +18,11 @@ import (
...
-	if port == "" {
-		log.Fatal("Missing: PORT environment variable")
-	}
-
-	// Grab $DATABASE_URL from env
-	databaseURL := os.Getenv("DATABASE_URL")
-	if databaseURL == "" {
-		log.Fatal("Missing: DATABASE_URL environment variable")
+	var config common.Config
+	if err := envconfig.Process("hercules", &config); err != nil {
```
Comment (hypnoglow) [inline server.go]: Just to note that previous variable name PORT is now HERCULES_PORT. The same is for DATABASE_URL, it is now HERCULES_DATABASEURL (no underscore between DATABASE and URL). If this is desired, then lgtm :)

**Comment by kshitij10496** [comment]
Thanks, @rafaelhenrique, and @hypnoglow for your invaluable contributions! 🎉 🤗

**Comment by rafaelhenrique** [comment]
:tada:

**Comment by kshitij10496** [comment]
UPDATE:

I deployed the changes to Heroku only to later realise that the platform sets the `PORT` environment variable for each application. What this means is that it won't update the new `HERCULES_PORT` variable. Thus, I had to [force the `envconfig` to recognize](https://github.com/kshitij10496/hercules/blob/77982c8ba231b6ad8f49597754b5db478218f1fa/common/config.go#L6) `PORT` and read it's value into `config.Port` variable. 😅

**Pull request merged**